theano/tensor/basic.py

```diff
@@ -5269,6 +5266,129 @@ def dot(a, b):
 # Linalg : TensorDot
 #########################
 
+def _tensordot_as_dot(a, b, axes, dot, batched):
+    """
+    Reduces a tensor dot product to a matrix or vector dot product. Based
+    on code from Tijmen Tieleman's gnumpy
+    (http://www.cs.toronto.edu/~tijmen/gnumpy.html).
+
+    Please see the documentation of tensordot for the meaning of the a, b
+    and axes arguments.
+
+    :param dot: a function that accepts two symbolic variables and computes
+                the appropriate dot product (e.g. dot, batched_dot)
+    :type dot: function
+
+    :param batched: whether to treat the first axis of a and b as a batch
+                    axis.  If so, this axis will be preserved in the output,
+                    allowing this function to be used also for batched
+                    tensor dot products.
+    :type batched: boolean
+
+    :returns: a tensor with shape equal to the concatenation of a's shape
+              (less any dimensions that were summed over) and b's shape
+              (less the first dimension and any dimensions that were summed
+              over).
+    :rtype: symbolic tensor
+    """
+    a, b = as_tensor_variable(a), as_tensor_variable(b)
+
+    if not numpy.isscalar(axes) and len(axes) != 2:
+        raise ValueError('Axes should be an integer or a '
+                         'list/tuple of len 2 (%s was provided)'
+                         % repr(axes))
```

Comment:
we use str(axes) everywhere else in Theano. For consistency continue to do this.